### PR TITLE
Change deprecated use of .astype() to .view()

### LINKED ISSUE #1261: pandas FutureWarning when casting type in DatetimeIndex
A new `FutureWarning` warning is being issued from pandas from `pvlib.solarposition.spa_python()`.

**Steps to reproduce**
```python
>>> import pandas as pd
>>> import pvlib
>>> ti = pd.DatetimeIndex([pd.to_datetime('today')])
>>> pvlib.solarposition.spa_python(ti, 46.0, 119.0)
```

**Observed behavior**
```python
.../lib/python3.9/site-packages/pvlib/solarposition.py:368: FutureWarning: casting datetime64[ns] values to int64 with .astype(...) is deprecated and will raise in a future version. Use .view(...) instead.
  unixtime = np.array(time.astype(np.int64)/10**9)
                            apparent_zenith  ...  equation_of_time
2021-07-15 09:58:51.008140        72.817018  ...         -6.010526
[1 rows x 6 columns]
```

**Expected behavior**
```python
                            apparent_zenith  ...  equation_of_time
2021-07-15 09:58:51.008140        72.817018  ...         -6.010526
[1 rows x 6 columns]
```

**Environment**
pvlib version: 0.8.1
pandas version: 1.3.0
Python version: 3.9.6
OS: Arch Linux (up-to-date)

**Additional information**
We are not using _pvlib_ directly; it is being pulled in by _pvfactors_. This issue started occurring when pandas was updated to 1.3.0.

May also be reproduced using `pytest pvlib/tests/test_solarposition.py`.

### PR DESCRIPTION
**Pull request by hashstat**
This change is made as suggested by the pandas FutureWarning to prevent
those warnings.

See #1261

 - [x] Closes #1261
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [ ] Tests added
 - [ ] Updates entries to [`docs/sphinx/source/api.rst`](https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/api.rst) for API changes.
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [ ] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [ ] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

### CODE REVIEW

File: docs/sphinx/source/whatsnew/v0.9.0.rst [hunk 1]
```diff
@@ -180,6 +180,8 @@ Bug fixes
   when data time step is not one minute. Error was introduced in v0.8.1.
   (:issue:`1241`, :pull:`1242`)
 * Corrected :py:func:`~pvlib.solarposition.spa_python` to avoid a future warning. (:pull:`1256`)
+* Changed deprecated use of ``.astype()`` to ``.view()`` in :py:mod:`~pvlib.solarposition`.
+  (:issue:`1261`, :pull:`1262`)
```
Comment (kanderso-nrel) [inline docs/sphinx/source/whatsnew/v0.9.0.rst:184]: Could merge this with the entry above it, nbd though

**Review by kanderso-nrel — APPROVED** [review]
LGTM, thanks @hashstat!

FYI to the other maintainers, looks like there's a handful of places in the tests that need this update as well:  https://dev.azure.com/solararbiter/pvlib%20python/_build/results?buildId=5277&view=logs&j=80999419-997d-54cc-7d9e-fafbf8111e17&t=fe7e5fe7-75d4-55e3-99ba-184521a271ef&l=232

File: docs/sphinx/source/whatsnew/v0.9.0.rst (hunk 1 above)
Comment (hashstat) [reply docs/sphinx/source/whatsnew/v0.9.0.rst]: Merged as suggested.

**Comment by wholmgren** [comment]
Thanks @hashstat!

**Pull request merged**